### PR TITLE
[FLINK-15907] [runtime] [configuration] expose getRawValue as protect…

### PR DESCRIPTION
**Pull request by stevenzwu**

### CODE REVIEW

**Comment by aljoscha** [comment]
@dawidwys What do you think about this one? I'm skeptical because making the method `protected` increases the API surface that we would need to keep stable? Shouldn't "modern" stuff go against `ReadableConfig`? 

Another option is to use `ReadableConfigToConfigurationAdapter` (which is `@Internal` right now) with an underlying `ReadableConfig`, which is a much simpler interface.

**Comment by dawidwys** [comment]
I agree with you in principle. I am also sceptical about this change for the same reasons. My take is that `Configuration` was certainly not designed to be extensible.

I think simply using `ReadableConfig` is not a solution to @stevenzwu use case as it is used only in a couple of locations. Majority of the code still uses Configuration. Similar approach to `ReadableConfigToConfigurationAdapter` could work but this has also limitation (that's why it is internal). It is not writable/it does not give access to all the keys/map of string based properties etc.). This would be my suggested way nevertheless. I think this can achieve the same goal as making the `getRawValue` method protected. It certainly requires more code on the user side, as basically the whole `Configuration` has to be reimplemented.

To sum up, I would not include this change as it creates a contract we rather don't want to maintain while the same goal can be achieved in a different way .

**Comment by aljoscha** [comment]
To make things easier, we could make the method protected but mark it as `@Internal`, meaning that we can still change it anytime we want.

**Comment by dawidwys** [comment]
I will not oppose but it would be weird from my point of view. We will have a protected method that is not used from our code from outside of the class.

This will be quite a puzzle for future maintainers. Also from the user perspective it's quite weird as we are saying, sure use this method with this commit, but we do not guarantee that the next commit will not break it for you.

**Comment by stevenzwu** [comment]
@dawidwys @aljoscha I actually didn't know `ReadableConfigToConfigurationAdapter` before you brought it up. would you be ok to expose it by removing the `@Internal` annotation?

**Comment by stevenzwu** [comment]
I might have spoken too soon. It seems like we do use the setters sometimes. I need to investigate if we can get rid of them via other way.

{code}
setInteger(RestOptions.PORT, restPort)
setString(JobManagerOptions.ADDRESS, hostname)
{code}

**Comment by stevenzwu** [comment]
Actually, we may not need this. We can work around by just transferring properties into Flink `Configuration` by calling `Configuration#setString(String key, String value) `

Closing this for now.
